glider_dac/glider_emails.py

```diff
@@ -62,3 +81,104 @@ def get_erddap_catalog_url():
     }
     url = 'http://%(host)s/erddap/metadata/iso19115/xml/' % args
     return url
+
+def glider_deployment_check(data_type=None, completed=True, force=False,
+                            deployment_dir=None, username=None):
+    """
+    """
+    cs = CheckSuite()
+    cs.load_all_available_checkers()
+    with app.app_context():
+        if data_type is not None:
+            is_delayed_mode = data_type == 'delayed'
+            if is_delayed_mode:
+                q_dict = {"delayed_mode": True,
+                          "completed": completed}
+            else:
+                q_dict = {"$or": [{"delayed_mode": False},
+                                  {"delayed_mode": {"$exists": False}}],
+                          "completed": completed}
+
+            if not force:
+                q_dict["compliance_check_passed"] = {"$ne": True}
+
+        # TODO: combine username/deployment cases?
+        if username:
+            q_dict = {"username": username}
+        # a particular deployment has been specified
+        elif deployment_dir:
+            q_dict = {"deployment_dir": deployment_dir}
+        else:
+            q_dict = {}
+
+        agg_pipeline = [{"$match": q_dict},
+                        {"$group": {"_id": "$user_id",
+                           "deployments": {"$push":
+                                {"_id": "$_id",
+                                 "name": "$name",
+                                 "deployment_dir": "$deployment_dir"} } }
+                        }]
+        # if force is enabled, re-check the datasets no matter what
+
+        # is this syntax still used?  if the first fn call fails, use the
+        # second set of results
+        try:
+            agg_result_set = db.deployments.aggregate(agg_pipeline)['result']
+        except:
+            agg_result_set = db.deployments.aggregate(agg_pipeline, cursor={})
+        for res in agg_result_set:
+            user = db.users.find_one(res["_id"])
+            all_messages = []
+            failing_deployments = []
+            for dep in res['deployments']:
+                root_logger.info("Running compliance check on glider "
+                                 "deployment: {}".format(dep))
+                try:
+                    dep_passed, dep_messages = process_deployment(dep)
+                    all_messages.append(dep_messages)
+                    if not dep_passed:
+                        failing_deployments.append(dep)
+                except Exception as e:
+                    root_logger.exception(
+                        "Exception occurred while processing deployment {}".format(dep['name']))
+                    text_body = ''
+            send_deployment_cchecker_email(user, failing_deployments,
+                                           "\n".join(all_messages))
+
+def process_deployment(dep):
```

Comment:
Can you add a docstrings to your functions? Looks like this function only runs the checker, maybe a better name would be:

`def check_compliance(deployment)`